src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,218 @@
+(jordan : ∀ a b : A, (a * b) * (a * a) = a * (b * (a *a)))
+
+/-- A (commutative) Jordan multiplication is also a Jordan multipication -/
+@[priority 100] -- see Note [lower instance priority]
+instance is_comm_jordan.to_is_jordan [has_mul A] [is_comm_jordan A] : is_jordan A :=
+{ lmul_comm_rmul := λ a b, by rw [is_comm_jordan.mul_comm, is_comm_jordan.mul_comm a b],
+  lmul_lmul_comm_lmul := λ a b, by rw [is_comm_jordan.mul_comm (a * a) (a * b),
+    is_comm_jordan.jordan, is_comm_jordan.mul_comm b (a * a)],
+  lmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm, ←is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm],
+  lmul_lmul_comm_rmul :=  λ a b, by rw [is_comm_jordan.mul_comm (a * a) (b * a),
+    is_comm_jordan.mul_comm b a, is_comm_jordan.jordan, is_comm_jordan.mul_comm,
+    is_comm_jordan.mul_comm b (a * a)],
+  rmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm b a, is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm], }
+
+
+/-- Semigroup multiplication satisfies the (non-commutative) Jordan axioms-/
+@[priority 100] -- see Note [lower instance priority]
+instance semigroup.is_jordan [semigroup A] : is_jordan A :=
+{ lmul_comm_rmul := λ a b, by rw mul_assoc,
+  lmul_lmul_comm_lmul := λ a b, by rw [mul_assoc, mul_assoc],
+  lmul_comm_rmul_rmul := λ a b, by rw [mul_assoc],
+  lmul_lmul_comm_rmul := λ a b, by rw [←mul_assoc],
+  rmul_comm_rmul_rmul := λ a b, by rw [← mul_assoc, ← mul_assoc], }
+
+@[priority 100] -- see Note [lower instance priority]
+instance comm_semigroup.is_comm_jordan [comm_semigroup A] : is_comm_jordan A :=
+{ mul_comm := mul_comm,
+  jordan := λ a b, mul_assoc _ _ _, }
+
+local notation `L` := add_monoid.End.mul_left
+local notation `R` := add_monoid.End.mul_right
+
+/-! The Jordan axioms can be expressed in terms of commuting multiplication operators -/
+section lie
+variables {A} [non_unital_non_assoc_ring A] [is_jordan A]
+
+@[simp] lemma lie_lmul_rmul (a : A) : ⁅L a, R a⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_comm_rmul _ _).symm
+
+@[simp] lemma lie_lmul_lmul_sq (a : A) : ⁅L a, L (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_lmul_comm_lmul _ _).symm
+
+@[simp] lemma lie_lmul_rmul_sq (a : A) : ⁅L a, R (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_comm_rmul_rmul _ _).symm
+
+@[simp] lemma lie_lmul_sq_rmul (a : A) : ⁅L (a * a), R a⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_lmul_comm_rmul _ _)
+
+@[simp] lemma lie_rmul_rmul_sq (a : A) : ⁅R a, R (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.rmul_comm_rmul_rmul _ _).symm
+
+end lie
+
+variables [non_unital_non_assoc_ring A] [is_comm_jordan A]
+
+/-- Linearise the operator form of the Jordan axiom (`lie_lmul_lmul_sq`) by substituting in
+a → a + b and expanding. -/
+lemma lie_lmul_lmul_add_add_add_eq_zero (a b : A) :
+  ⁅L a, L (b * b)⁆ + ⁅L b, L (a * a)⁆ + 2•⁅L a, L (a * b)⁆ + 2•⁅L b, L (a * b)⁆ = 0 :=
```

Comment:
We should state this in the more symmetric form:
```suggestion
  ⁅L a, L (b * b)⁆ + ⁅L b, L (a * a)⁆ + 2•⁅L a, L (a * b)⁆ + 2•⁅L b, L (b * a)⁆ = 0 :=
```
In fact I suspect the following would be the most useful form to state this lemma:
```lean
  2•(⁅L a, L (a * b)⁆ + ⁅L b, L (b * a)⁆) = ⁅L (a * a), L b⁆ + ⁅L (b * b), L a⁆ :=
```
What do you think?

Reply:
Done.